### PR TITLE
Layerlist toolbar

### PR DESCRIPTION
**Pull request by pelord**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines:https://github.com/infra-geo-ouverte/igo2/blob/master/.github/CONTRIBUTING.md#git-commit-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Layer-list toolbar button were not linked to the layers status (visibility or resolution).

**What is the new behavior?**
Fix the non-synchronized status.
+
Observable on 
- layer.visible as layer.visible$
- layer.isInResolutionsRange as layer.isInResolutionsRange$

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications:

**Other information**:

### CODE REVIEW

File: packages/geo/src/lib/layer/shared/layers/layer.ts [hunk 1]
```diff
@@ -51,12 +51,14 @@ export abstract class Layer {
     this.options.baseLayer = baseLayer;
   }
 
+  readonly visible$: BehaviorSubject<boolean> = new BehaviorSubject(undefined);
```
Comment (cbourget) [inline packages/geo/src/lib/layer/shared/layers/layer.ts]: This is good.

File: packages/geo/src/lib/layer/layer-list/layer-list-binding.directive.ts [hunk 2]
```diff
@@ -26,20 +29,44 @@ export class LayerListBindingDirective implements OnInit, AfterViewInit, OnDestr
...
     // Override input layers
     this.component.layers = [];
-
-    this.layers$$ = this.mapService
-      .getMap()
-      .layers$.subscribe((layers: Layer[]) => {
-        this.component.layers = layers.filter((layer: Layer) => {
-          return layer.showInLayerList === true;
-        });
+    this.layersOrResolutionChange$$ = combineLatest([
```
Comment (cbourget) [inline packages/geo/src/lib/layer/layer-list/layer-list-binding.directive.ts]: This works but is not ideal. See the last comment of this review. Also, settting the layers eveytime the resolution changes is not needed.

Comment (cbourget) [reply packages/geo/src/lib/layer/layer-list/layer-list-binding.directive.ts]: Actually, does it work? isInResolutionsRange$ is never updated unless it's getter is invoked and I don't see that anywhere.

**Comment by pelord** [comment]
@cbourget  Any news about this PR? Did you had time to work on this?

**Comment by pelord** [comment]
I'll try to pull a patch based on the method used by the legacy branch until you fix this branch? Ok?

**Comment by cbourget** [comment]
Alright. I'll try and get back to it soon.

**Pull request merged**